### PR TITLE
Michaelberk/ch212265/sample size additions

### PR DESCRIPTION
**Pull request by micahel2B**
## Which problem does this PR solve?
- Increase sensitivity of calculator
- Standardize metrics to `experiment_user_data` macro
- Add series and movie tvt metrics
## Main changes
- See above PR
## How did I test these changes?
- Ran each metric and observed results
## Related documents/resources
[Ticket](https://app.clubhouse.io/tubi/story/212265/sample-size-additions)
## Note
Planned next steps are:
1. Merge PR
2. Pull updated code to sample size calculator
3. Remove dev directory

If we expect to make many more edits, it would make sense to create a `dev` and `production` folder within the `sample_size_calculator` folder in TDR. Not high urgency or priority though.

### CODE REVIEW

File: ssc_utils/raw_user_data.py [hunk 1]
```diff
@@ -10,17 +10,21 @@ def generate_raw_user_data_cte(self, prev_cte_sql):
               SELECT 
                   a.device_id,
                   device_first_seen_ts,
+                  device_first_view_ts,
```
Comment (bzou-tubi) [inline ssc_utils/raw_user_data.py:13]: You'll need to add an extra column to the group by down below

File: ssc_utils/metric_switcher.py [hunk 2]
```diff
@@ -187,14 +187,14 @@ def ad_impressions(self):
...
-                   d.first_exposure_ds,
-                   SUM(COALESCE(rev.ad_impression_total_count, 0))::float AS ad_impression_total_count,
-                   SUM(COALESCE(rev.gross_revenue, 0))::float AS gross_revenue
+            SELECT  d.device_id,
+                    d.ds,
+                    d.platform,
+                    d.platform_type,
+                    d.device_first_seen_ts,
+                    d.first_exposure_ds,
+                    SUM(rev.ad_impression_total_count) AS ad_impression_total_count,
```
Comment (bzou-tubi) [inline ssc_utils/metric_switcher.py:196]: Why remove the coalesce and the cast to float? Did you test to make sure the results stay the same?

`tubidw.experiment_samplesize_input` has the same

File: ssc_utils/metric_switcher.py [hunk 3]
```diff
@@ -14,95 +14,95 @@ def generate_user_data_cte(self, metric): # <-- whatever you input here in "metr
...
+            'tvt-capped_new_visitors',
+            'registration-did_signup',
+            'registration-did_activate', 
             'visits',
-            'viewer_conversion',
-            'new_viewer_first_day_conversion',
-            'new_user_1_to_8_days_retained',
-            'all_user_retained_in_experiment_timeframe',
+            'conversion',
+            'conversion--new_visitors',
```
Comment (bzou-tubi) [inline ssc_utils/metric_switcher.py:34]: Same with these, we have the 5 minute versions in `primary_metrics`. Can you add those and remove these (if you're not using them)?

File: ssc_utils/metric_switcher.py [hunk 4]
```diff
@@ -112,61 +112,61 @@ def visits(self):
...
           SELECT 
             device_id, ds, platform_type, platform, device_first_seen_ts, first_exposure_ds, 
-            'new_user_1_to_8_days_retained'::text AS metric_name,
+            'retention--new_viewers'::text AS metric_name,
             'MAX'::text AS metric_collection_method, 
-            CASE WHEN ds > device_first_seen_ts + INTERVAL '1 day' AND tvt_sec > 10 THEN 1.0 ELSE 0.0 END AS metric_value 
+            CASE WHEN ds > device_first_view_ts + INTERVAL '1 day' AND (tvt_sec + linear_tvt_sec) > 10 THEN 1.0 ELSE 0.0 END AS metric_value
           FROM raw_user_data
-          WHERE ds >= DATE_TRUNC('day', device_first_seen_ts) AND ds < device_first_seen_ts + INTERVAL '8 day'
+          WHERE ds >= DATE_TRUNC('day',device_first_seen_ts) AND ds < device_first_seen_ts + INTERVAL '7 day'
```
Comment (bzou-tubi) [inline ssc_utils/metric_switcher.py:156]: I think this should use `device_first_view_ts` too

**Review by bzou-tubi — CHANGES_REQUESTED** [review]
Handful of small comments. 

Also, there's no need to stage/commit the `ipynb_checkpoints` and `__pycache__` files, can you please remove them from this PR? Tip to avoid this next time: do a `git status` to look at which specific files need to be staged. Unfortunately this is necessary for ipynb notebooks because they auto create these checkpoint files

File: ssc_utils/metric_switcher.py (hunk 2 above)
Comment (micahel2B) [reply ssc_utils/metric_switcher.py:196]: Good spot.

**Review by bzou-tubi — APPROVED** [review]
LGTM

**Pull request merged**